backend/operations_scanner.go

```diff
@@ -660,17 +665,18 @@ func (s *OperationsScanner) postAsyncNotification(ctx context.Context, op operat
 // convertClusterStatus attempts to translate a ClusterStatus object from
 // Cluster Service into an ARM provisioning state and, if necessary, a
 // structured OData error.
-func convertClusterStatus(clusterStatus *arohcpv1alpha1.ClusterStatus, current arm.ProvisioningState) (arm.ProvisioningState, *arm.CloudErrorBody, error) {
+func (s *OperationsScanner) convertClusterStatus(ctx context.Context, logger *slog.Logger,
+	clusterStatus *arohcpv1alpha1.ClusterStatus, current arm.ProvisioningState,
+	internalId ocm.InternalID) (arm.ProvisioningState, *arm.CloudErrorBody, error) {
 	var opStatus = current
 	var opError *arm.CloudErrorBody
 	var err error
 
 	switch state := clusterStatus.State(); state {
 	case arohcpv1alpha1.ClusterStateError:
 		opStatus = arm.ProvisioningStateFailed
-		// FIXME This is guesswork. Need clarity from Cluster Service
```

Comment:
@mbarnes, I added a reference to the cluster service repo. Does this meet your expectations?

Reply:
That helps, thanks.  I'm still not sure if it's worth calling `clusterStatus.Description()` as a fallback.  Would that ever return a message appropriate for error details?  `ProvisionErrorCode` and `ProvisionErrorMessage` seem like the right things to check for a failure during provisioning, but what about a failed cluster update?

Reply:
Going through CS code mappers, it seems reasonable to use description as a fallback https://gitlab.cee.redhat.com/service/uhc-clusters-service/-/blob/master/pkg/mappers/outbound/clusters.go?ref_type=heads#L787 (if both are empty, a generic error message is set).

Can you please elaborate on a failed cluster update in this context? Is the flow for cluster update also trigger `convertClusterStatus`?